os/drivers/audio/alc5658.c

```diff
@@ -974,8 +961,35 @@ static int alc5658_ioctl(FAR struct audio_lowerhalf_s *dev, int cmd, unsigned lo
 	switch (cmd) {
 
 	case AUDIOIOC_PREPARE: {
-		/* TODO: Need to be implemented later */
-		audvdbg("AUDIOIOC_PREPARE: No Action Would be taken now \n");
+		audvdbg("AUDIOIOC_PREPARE: alc5658 prepare\n");
+		int entry;
+
+		/* Take semaphore */
+		alc5658_takesem(&priv->devsem);
+
+		/* Pause i2s channel */
+		I2S_PAUSE(priv->i2s);
+
+		/*Reconfigure alc5658 */
+		/* Set first set of registers */
+		alc5658_exec_i2c_script(priv, codec_init_inout_script1, sizeof(codec_init_inout_script1) / sizeof(t_codec_init_script_entry));
+
+		/* Get configured sample rate and set it here - valdiation done in configure itself */
+		entry = alc5658_get_sample_rate_script(priv->samprate);
+		alc5658_exec_i2c_script(priv, g_sample_entry[entry].script, g_sample_entry[entry].size / sizeof(t_codec_init_script_entry));
+
+		/* Set second set of registers */
+		alc5658_exec_i2c_script(priv, codec_init_inout_script2, sizeof(codec_init_inout_script2) / sizeof(t_codec_init_script_entry));
+
+		/* TOCHECK: Possible to cut the two level execution of alc scritps so as to cut the time? */
+		alc5658_setregs(priv);
+		alc5658_getregs(priv);
```

Comment:
when it configured, getregs will be called.
Hence, information is enough. let's remove getregs here.

Reply:
getregs will be executed only when audio logs, are enabled (for debugging purpose). static void alc5658_getregs(struct alc5658_dev_s *priv)
{
	audvdbg("MIC GAIN 0x%x\n", (uint32_t)alc5658_readreg(priv, ALC5658_IN1));
	audvdbg("MUTE HPOUT MUTE %x\n", (uint32_t)alc5658_readreg(priv, ALC5658_HPOUT));
	audvdbg("VOLL 0x%x\n", (uint32_t)alc5658_readreg(priv, ALC5658_HPOUT_L));
	audvdbg("VOLR 0x%x\n", (uint32_t)alc5658_readreg(priv, ALC5658_HPOUT_R));
}
NOw, the conriguration happens only in prepare ioclt.